manager/manager/typeresolver.go

```diff
@@ -40,26 +39,23 @@ type TypeResolver interface {
 }
 
 type typeResolver struct {
-	getter  util.HTTPClient
 	maxUrls int
 	rp      registry.RegistryProvider
+	c       util.HTTPClient
+}
+
+type fetchableURL struct {
+	registry registry.Registry
```

Comment:
Since it's a struct field, it won't overload because it's always namespace to the struct: `foo.registry`.